### ISSUE TITLE
Add HTU21D temperature & humidity sensor

### ISSUE THREAD

**Issue by Novocaine85** [ESPresense/ESPresense#100]
**Describe the solution you'd like**
HTU21D sensor is used on ESP32-mqtt-room project. It would be cool, since looks like that project is no longer active, if you add support for this sensor like ESP32-mqtt-room. As per pull #54 only DHT11 and the DHT22 are supported.

**Comment by Novocaine85** [comment]
Hello!

Any news about this?

**Comment by DTTerastar** [comment]
It's gotten farther down the list, sorry.

**Comment by Novocaine85** [comment]
No rush. Cause I changed sensors meanwhile 😂

**Comment by Krytten** [comment]
Is this sensor still unsupported? I have a few HTU21D sensors running with 'ESP32-mqtt-room', but am looking at making the switch over to ESPresence as that it is now depreciated. I've looked in the documentation but don't see it mentioned.

**Comment by DTTerastar** [comment]
> Is this sensor still unsupported? I have a few HTU21D sensors running with 'ESP32-mqtt-room', but am looking at making the switch over to ESPresence as that it is now depreciated. I've looked in the documentation but don't see it mentioned.

Yeah, If I ordered its in this pile of stuff and I need to find it...

**Comment by CV8R** [comment]
Ended up here searching to see if the GY-21 module I found in my box is supported. I would send you one but the shipping will cost more than the module!

**Comment by CV8R** [comment]
> searching to see if the GY-21 module I found in my box is supported

Turns out it just works!

Plugged one in, set the I2C Bus 1 pins, then added "1" to I2C Bus for SHTC1/3, SHTW1/2, SHT3x/4x, SHT85 - Temperature and Humidity.

Logs show

```
I2C Bus 1:    sda=13 scl=16
SHT:          Auto-detect on bus 1
```

Temperature and Humidity are being sent over MQTT for the room.

I am not sure if the heater is enabled or not?  Doesn't matter for my use case - its only there to ensure that the device doesn't overheat in its enclosure.

**Comment by DTTerastar** [comment]
Can anyone test the attached PR via the alpha firmware installer and report back?

**Comment by ghost** [comment]
Thanks again for the request — HTU21D support is now in PR #2056 and available in the latest alpha build.\n\nCould you please test it via https://espresense.com/alpha and report back whether your HTU21D readings are working as expected (temp/humidity + your board/model + I2C address)?\n\nIf anything looks off, please share logs/screenshots and I’ll follow up quickly.

**Comment by DTTerastar** [comment]
Resolved: HTU21D already works — no firmware change was ever needed.

The existing **SHT** sensor module uses `arduino-sht`'s auto-detect, which probes `SHT2X`. That driver is protocol-identical to the HTU21D: I2C address `0x40`, commands `0xF3`/`0xF5` (no-hold), CRC-8 poly 0x31 init 0, and the exact HTU21D datasheet conversions (`T = -46.85 + 175.72·x/65536`, `RH = -6 + 125·x/65536`). Same for the Si7021.

That's exactly what @CV8R hit with a GY-21 — it just worked.

**To use an HTU21D / Si7021 / SHT21 / GY-21:**
1. Set your I2C Bus 1 (or 2) SDA/SCL pins.
2. Set **I2C Bus** under the SHT temperature/humidity section to `1` (or `2`).
3. Temperature and humidity publish to the room's `temperature` / `humidity` topics.

The only real bug was that the setting's label never said "HTU21D", so nobody could find it. Fixed in #2408. PR #2056 closed as redundant.

(@CV8R — on the heater: the SHT2x driver never enables it, so it's off.)
